spec/factories/middleware_datasource.rb

```diff
@@ -0,0 +1,18 @@
+FactoryGirl.define do
+  factory :middleware_datasource do
+  end
+
+  factory :hawkular_middleware_datasource,
+          :aliases => ['app/models/manageiq/providers/hawkular/middleware_manager/middleware_datasource'],
+          :class   => 'ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareDatasource',
+          :parent  => :middleware_datasource do
+  end
+
+  factory :hawkular_middleware_datasource_initialized,
```

Comment:
can you use hawkular_middleware_datasource as a parent and not copy paste some of the definitions here below?